sql/information_schema/information_schema.go

```diff
@@ -625,6 +625,115 @@ func charsetRowIter(ctx *Context, c Catalog) (RowIter, error) {
 	return RowsToRowIter(rows...), nil
 }
 
+func statisticsRowIter(ctx *Context, c Catalog) (RowIter, error) {
+	var rows []Row
+	dbs := c.AllDatabases(ctx)
+
+	for _, db := range dbs {
+		tableNames, err := db.GetTableNames(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, tableName := range tableNames {
+			tbl, _, err := c.Table(ctx, db.Name(), tableName)
+			if err != nil {
+				return nil, err
+			}
+
+			// TODO: Doesn't correctly consider primary keys from table implementations that don't implement sql.IndexedTable
```

Comment:
Not sure why you can't support this case rn. You can check whether a table just implements PrimaryKeyTable or Indexed table and proceed from there.